src/test/java/com/atlassian/db/replica/api/TestDualConnection.java

```diff
@@ -267,7 +268,7 @@ public void shouldDetectWriteOperation() throws SQLException {
 
         assertThat(connectionProvider.getProvidedConnectionTypes())
             .containsOnly(MAIN);
-        verify(databaseCall).call(any(), eq(new RouteDecisionBuilder(Reason.WRITE_OPERATION).sql(sql).build()));
+        verify(databaseCall).call(any(), eq(new RouteDecision(WRITE_OPERATION).withSql(sql)));
```

Comment:
Why does the `shouldDetectWriteOperation` expect a write operation? I don't see it from the `// given` section

Reply:
Ok I see, it's a function call